pkg/kubelet/hyper/hyperclient.go

```diff
@@ -343,39 +343,40 @@ func (client *HyperClient) GetPodIDByName(podName string) (string, error) {
 }
 
 func (client *HyperClient) ListPods() ([]HyperPod, error) {
-	v := url.Values{}
-	v.Set(KEY_ITEM, TYPE_POD)
-	body, _, err := client.call("GET", "/list?"+v.Encode(), "", nil)
-	if err != nil {
-		return nil, err
-	}
+	request := grpctypes.PodListRequest{}
 
-	var podList map[string]interface{}
-	err = json.Unmarshal(body, &podList)
+	response, err := client.client.PodList(context.Background(), &request)
 	if err != nil {
 		return nil, err
 	}
 
 	var result []HyperPod
-	for _, pod := range podList["podData"].([]interface{}) {
-		fields := strings.Split(pod.(string), ":")
+	for _, pod := range response.PodList {
+
 		var hyperPod HyperPod
-		hyperPod.PodID = fields[0]
-		hyperPod.PodName = fields[1]
-		hyperPod.VmName = fields[2]
-		hyperPod.Status = fields[3]
-
-		values := url.Values{}
-		values.Set(KEY_POD_NAME, hyperPod.PodID)
-		body, _, err = client.call("GET", "/pod/info?"+values.Encode(), "", nil)
+		hyperPod.PodID = pod.PodID
+		hyperPod.PodName = pod.PodName
+		hyperPod.VmName = pod.VmID
+		hyperPod.Status = pod.Status
+
+		req := grpctypes.PodInfoRequest{}
+
+		res, err := client.client.PodInfo(context.Background(), &req)
+		if err != nil {
+			return nil, err
+		}
+
+		body, err := json.Marshal(res.PodInfo)
```

Comment:
Just as the previous comment said, you should never do json marshal/unmarshal in this client, just return grpc type and check any places called this function to make sure the returned value can be consumed properly.

Using gRPC means you are hoping to call remote codes "locally", instead of HTTP -> json -> object way. 

cc @YaoZengzeng